--- a/src/cpp/session/SessionOptions.cpp	2024-05-07 06:31:08.000000000 +1200
+++ b/src/cpp/session/SessionOptions.cpp	2024-05-21 19:33:46.308996710 +1200
@@ -709,7 +709,7 @@
                               std::string* pPath)
 {
 #if defined(__linux__) && !defined(RSTUDIO_PACKAGE_BUILD)
-   FilePath dependenciesPath = resourcePath.completePath("../../dependencies/common/node/18.18.2");
+   FilePath dependenciesPath = resourcePath.completePath("/usr");
    resolvePath(dependenciesPath, pPath);
 #else
    resolvePath(resourcePath, pPath);
