diff --git a/packages/app-desktop/gui/NoteEditor/NoteBody/CodeMirror/utils/useContextMenu.ts b/packages/app-desktop/gui/NoteEditor/NoteBody/CodeMirror/utils/useContextMenu.ts
index 6289d84..580df52 100644
--- a/packages/app-desktop/gui/NoteEditor/NoteBody/CodeMirror/utils/useContextMenu.ts
+++ b/packages/app-desktop/gui/NoteEditor/NoteBody/CodeMirror/utils/useContextMenu.ts
@@ -1,5 +1,5 @@
 
-import { ContextMenuEvent, ContextMenuParams } from 'electron';
+import { ContextMenuParams, Event } from 'electron';
 import { useEffect, RefObject } from 'react';
 import { _ } from '@joplin/lib/locale';
 import { PluginStates } from '@joplin/lib/services/plugins/reducer';
@@ -69,7 +69,7 @@ const useContextMenu = (props: ContextMenuProps) => {
 			return intersectingElement && isAncestorOfCodeMirrorEditor(intersectingElement);
 		}
 
-		async function onContextMenu(event: ContextMenuEvent, params: ContextMenuParams) {
+		async function onContextMenu(event: Event, params: ContextMenuParams) {
 			if (!pointerInsideEditor(params)) return;
 
 			// Don't show the default menu.
